### PR TITLE
[bndtools.m2e] junit launcher doesn't understand maven profiles

### LINKED ISSUE #3576: [bndtools.m2e] junit launcher doesn't understand maven profiles
Eclipse M2E supports maven profiles. You can set a profile in the project configuration

Screenshot from 2019-11-15 21-18-50

But the bndtools launchers don't understand this and so do not show the launcher shortcuts.

### PR DESCRIPTION
**Pull request by rotty3000**
fixes #3576

Signed-off-by: Raymond Augé <raymond.auge@liferay.com>

### CODE REVIEW

File: bndtools.core/_plugin.xml [hunk 1]
```diff
@@ -605,9 +605,13 @@
...
                            <!--  Java element (src-folder, package or JUnit class) -->
                            <adapt type="org.eclipse.jdt.core.IJavaElement">
-                              <test property="org.eclipse.jdt.core.hasTypeOnClasspath" value="junit.framework.Test" />
-                              <test property="org.eclipse.jdt.junit.canLaunchAsJUnit" forcePluginActivation="true" />
                               <test property="org.bndtools.core.isInBndJavaProject" forcePluginActivation="true" />
+                              <test property="org.eclipse.jdt.junit.canLaunchAsJUnit" forcePluginActivation="true" />
+                              <or>
+                                 <test property="org.eclipse.jdt.core.hasTypeOnClasspath" value="org.junit.Test" />
+                                 <test property="org.eclipse.jdt.core.hasTypeOnClasspath" value="junit.framework.Test" />
+                                 <test property="org.eclipse.jdt.core.hasTypeOnClasspath" value="org.junit.jupiter.api.Test" />
```
Comment (bjhargrave) [inline bndtools.core/_plugin.xml]: There are more Junit 5 types to consider. You may need to map this `<test>` to a set of `<tests>`:

https://github.com/bndtools/bnd/blob/9c99bbd913df0c2c78114868b61da4d3d50e0fea/maven/bnd-maven-plugin/src/main/java/aQute/bnd/maven/plugin/TestCases.java#L7-L11

File: bndtools.m2e/_plugin.xml [hunk 2]
```diff
@@ -52,6 +52,7 @@
 							<or>
 								<test property="org.eclipse.jdt.core.hasTypeOnClasspath" value="org.junit.Test" />
 								<test property="org.eclipse.jdt.core.hasTypeOnClasspath" value="junit.framework.Test" />
+								<test property="org.eclipse.jdt.core.hasTypeOnClasspath" value="org.junit.jupiter.api.Test" />
```
Comment (bjhargrave) [inline bndtools.m2e/_plugin.xml]: Need more junit 5 tests in `<or>` as in the above comment.

File: bndtools.core/_plugin.xml (hunk 1 above)
Comment (rotty3000) [reply bndtools.core/_plugin.xml]: I don't think so. Please not this is not a check of the classes and how they are annotated, it's just checking if a class is on the classpath, Which means you only need a single one from each junit framework version.

File: bndtools.m2e/_plugin.xml (hunk 2 above)
Comment (rotty3000) [reply bndtools.m2e/_plugin.xml]: No I don't think so.

File: bndtools.core/_plugin.xml (hunk 1 above)
Comment (bjhargrave) [reply bndtools.core/_plugin.xml]: OK

**Pull request merged**